src/Data/Fin/Properties.agda

```diff
@@ -364,14 +364,24 @@ lower₁-irrelevant {suc n} (suc i)  _   _ =
 ------------------------------------------------------------------------
 -- inject≤
 
-toℕ-inject≤ : ∀ {m n} (i : Fin m) (le : m ℕ≤ n) →
+toℕ-inject≤ : ∀ {m n} (i : Fin m) .(le : m ℕ≤ n) →
                 toℕ (inject≤ i le) ≡ toℕ i
-toℕ-inject≤ zero    (s≤s le) = refl
-toℕ-inject≤ (suc i) (s≤s le) = cong suc (toℕ-inject≤ i le)
-
-inject≤-refl : ∀ {n} (i : Fin n) (n≤n : n ℕ≤ n) → inject≤ i n≤n ≡ i
-inject≤-refl zero    (s≤s _  ) = refl
-inject≤-refl (suc i) (s≤s n≤n) = cong suc (inject≤-refl i n≤n)
+toℕ-inject≤ {_} {suc n} zero    _  = refl
+toℕ-inject≤ {_} {suc n} (suc i) le = cong suc (toℕ-inject≤ i (ℕ.≤-pred le))
+
+inject≤-refl : ∀ {n} (i : Fin n) .(n≤n : n ℕ≤ n) → inject≤ i n≤n ≡ i
+inject≤-refl {suc n} zero    _   = refl
+inject≤-refl {suc n} (suc i) n≤n = cong suc (inject≤-refl i (ℕ.≤-pred n≤n))
+
+inject≤-irrelevant : ∀ {m n} (i : Fin m) (m≤n₁ m≤n₂ : m ℕ≤ n)
+                     → inject≤ i m≤n₁ ≡ inject≤ i m≤n₂
+inject≤-irrelevant {_} {suc n} zero m≤n₁ m≤n₂    = refl
+inject≤-irrelevant {_} {suc n} (suc i) m≤n₁ m≤n₂ = refl
+
+inject≤-idempotent : ∀ {m n k} (i : Fin m) (m≤n : m ℕ≤ n) (n≤k : n ℕ≤ k) (m≤k : m ℕ≤ k)
+                     → inject≤ (inject≤ i m≤n) n≤k ≡ inject≤ i m≤k
+inject≤-idempotent {_} {suc n} {suc k} zero _ _ _ = refl
+inject≤-idempotent (suc i) (s≤s m≤n) (s≤s n≤k) (s≤s m≤k) = cong suc (inject≤-idempotent i m≤n n≤k m≤k)
 
 ------------------------------------------------------------------------
```

Comment:
This ought to work too:

```agda
inject≤-irrelevant : ∀ {m n} (i : Fin m) (m≤n₁ m≤n₂ : m ℕ≤ n)
                     → inject≤ i m≤n₁ ≡ inject≤ i m≤n₂
inject≤-irrelevant _ _ _ = refl

inject≤-idempotent : ∀ {m n k} (i : Fin m) .(m≤n : m ℕ≤ n) .(n≤k : n ℕ≤ k) .(m≤k : m ℕ≤ k)
                     → inject≤ (inject≤ i m≤n) n≤k ≡ inject≤ i m≤k
inject≤-idempotent {_} {suc n} {suc k}     zero    _ _ _ = refl
inject≤-idempotent {n = suc n} {k = suc k} (suc i) _ _ _ =
  cong suc (inject≤-idempotent i _ _ _)
```

Reply:
Sorry, that was rather silly of me, not to notice that ```inject\le-irrelevant``` is not necessary at all.
I do not spend too much time on this. I hope it is merged soon.